gpflow/kernels/base.py

```diff
@@ -36,25 +36,28 @@ def __init__(self,
                  active_dims: Optional[Union[slice, list]] = None,
                  name: Optional[str] = None):
         """
-        :param active_dims: active dimensions, has the slice type.
+        :param active_dims: active dimensions, either a slice or list of
+            indices into the columns of X.
         :param name: optional kernel name.
         """
         super().__init__(name=name)
-        if isinstance(active_dims, list):
-            active_dims = np.array(active_dims)
-        self._active_dims = active_dims
+        self._active_dims = self._normalize_active_dims(active_dims)
+
+    @staticmethod
+    def _normalize_active_dims(value):
+        if value is None:
+            value = slice(None, None, None)
+        if not isinstance(value, slice):
+            value = np.array(value, dtype=int)
```

Comment:
```suggestion
            value = tf.constant(value, dtype=tf.int32)
```

Reply:
What's the advantage of having it be a tensor? Why had you made it a numpy array before?

Reply:
The same reason as to why `tf.function` should receive only tensors - it is a safety precaution from graph regeneration.

Reply:
It's kinda weird though that the `Kernel` constructor specifies numpy array, and then it gets turned into a tensor...?

Reply:
No, it is not weird, it is internal *only* valid representation. When you pass ndarray to the parameter, you don't mind for it being converted to tf.variable ;)

Reply:
True. Was there any reason behind your choice to originally make it a np.array()? (Just trying to clarify/understand because you're asking me to change a line of code that is actually unrelated to the rest of my PR's content, and it's one that you authored in the first place.)